persistent/persistence.py

```diff
@@ -315,7 +337,15 @@ def __setstate__(self, state):
                 raise TypeError('No instance dict')
             idict.clear()
             for k, v in inst_dict.items():
-                idict[intern(k)] = v
+                # Normally the keys for instance attributes are interned.
+                # Do that here, but only if it is possible to do so.
+
+                # TODO: On Python 2 codebases that straddle Python3,
+                # and use 'from __future__ import unicode_literals' it's not
+                # unheard of to have unicode objects in the __dict__ by accident.
+                # Should we watch for that and attempt to encode it so it can be
+                # interned?
```

Comment:
I wouldn't make allowances for the future import (I consider that particular one a bug injector).